### PR TITLE
Add to core concepts in guide

### PR DESCRIPTION
**Pull request by moradology**
This PR adds some introductory discussion for the `LayoutScheme` and `LayoutDefinition` core concepts

### CODE REVIEW

**Comment by lossyrob** [comment]
@moradology does this need to be updated, or does the core concepts already cover everything

**Comment by fosskers** [comment]
I like this PR but I'm confused where the conflicts are coming from. Too bad we can't view them right in Github.

**Comment by moradology** [comment]
Doing the rebase on it right now + adding a bit of detail about `FloatingLayoutScheme` and anything that's empty

**Comment by fosskers** [comment]
Reviewing.

File: docs/guide/core-concepts.md [hunk 1]
```diff
@@ -934,25 +933,65 @@ Here are some useful `Extent` operations, among many more:
...
 Layout Definitions describes that grid's area and cell width/height.
+These definitions can be used to chop a bundle of imagery into tiles
+suitable for being served out on a web map.  
 
 Within the context of GeoTrellis, the `LayoutDefinition` class extends
 `GridExtent`, and exposes methods for querying the sizes of the grid and
 grid cells. Those values are stored in the `TileLayout` (the grid
 description) and `CellSize` classes respectively. `LayoutDefinition`s are
 used heavily during the raster reprojection process.
+||||||| merged common ancestors
```
Comment (fosskers) [inline docs/guide/core-concepts.md]: A funky leftover?

File: docs/guide/core-concepts.md [hunk 2]
```diff
@@ -1398,9 +1374,73 @@ trait Functor[F[_], A] extends MethodExtensions[F[A]]{
 
 which allows us to do:
 
-```scala
+``scala
```
Comment (fosskers) [inline docs/guide/core-concepts.md]: Missing backtick here.

Comment (moradology) [inline docs/guide/core-concepts.md]: Good catch

**Comment by fosskers** [comment]
Looks good!

**Pull request merged**